packages/web-app-files/src/components/SideBar/Shares/Links/CreateQuickLink.vue

```diff
@@ -31,37 +41,45 @@ export default defineComponent({
     }
   },
   emits: ['createPublicLink'],
-  setup() {
+  setup(props, { emit }) {
     const { can } = useAbility()
+    const { $gettext } = useGettext()
+
     const canCreatePublicLinks = computed(() => can('create-all', 'PublicLink'))
-    return { canCreatePublicLinks }
-  },
-  computed: {
-    heading() {
-      return this.$gettext('Quick link')
-    },
-    createLinkHint() {
-      return this.$gettext('Create quick link')
-    },
-    createLinkLabel() {
-      return this.$gettext('Create link')
-    },
-    noLinkLabel() {
-      return this.$gettext('No link')
-    }
-  },
-  methods: {
-    createQuickLink() {
-      this.$emit('createPublicLink', {
+    const resource = inject<Resource>('resource')
+    const capabilities = inject<Capabilities>('capabilities')
+    const heading = computed(() => $gettext('Quick link'))
+    const createLinkHint = computed(() => $gettext('Create quick link'))
+    const createLinkLabel = computed(() => $gettext('Create link'))
+    const noLinkLabel = computed(() => $gettext('No link'))
```

Comment:
When you refactor strings / labels: it's fine now to use `$gettext` calls directly in the template. If the string construction logic gets too complex a computed prop is the way to go. But for such a simple `label` string I'd prefer inline usage in the template now :-)

Reply:
alright - sounds good to me!